shared/settings/advanced/container.js

```diff
@@ -10,18 +10,25 @@ import {
 import * as RouteTreeGen from '../../actions/route-tree-gen'
 import {HeaderHoc} from '../../common-adapters'
 import * as Constants from '../../constants/settings'
+import {anyErrors, anyWaiting} from '../../constants/waiting'
 import {compose} from 'recompose'
 import Advanced from './index'
 import {connect, lifecycle} from '../../util/container'
 
 type OwnProps = {||}
-const mapStateToProps = state => ({
-  hasRandomPW: !!state.settings.passphrase.randomPW,
-  lockdownModeEnabled: state.settings.lockdownModeEnabled,
-  openAtLogin: state.config.openAtLogin,
-  processorProfileInProgress: Constants.processorProfileInProgress(state),
-  traceInProgress: Constants.traceInProgress(state),
-})
+const mapStateToProps = state => {
+  const settingLockdownMode = anyWaiting(state, Constants.setLockdownModeWaitingKey)
+  const setLockdownModeError = anyErrors(state, Constants.setLockdownModeWaitingKey)
```

Comment:
I couldn't get this to be set using the link conditioner, for me the page stays like this if I open it with no internet:
<img width="357" alt="image" src="https://user-images.githubusercontent.com/11968340/51993008-07fbd500-247c-11e9-84e2-69c92246f9a5.png">
expected?

Reply:
Ah, yes, you can repro it by going to the page with internet, turning off the internet, and then trying to toggle the switch.

A separate issue is that if you go to this page without internet, and then turn it on, the toggle stays disabled. Is it worth using the gregor connection state to re-enable it? What do you think?

Reply:
That sounds like something that `Reloadable` could handle?

Reply:
It would take over the whole screen, which isn't great -- and the user would still have to retry by clicking 'Reload'.